data_handler/handlers/helpers.py

```diff
@@ -72,6 +75,104 @@ def set_value(self, token: str, value: decimal.Decimal):
+            self, token_name: str, current_block: int, collateral_change: Decimal, debt_change: Decimal
+    ) -> None:
+        """
+        Update the state of interest rate calculation with the new data.
+        :param token_name: str - The name of the token, for example `STRK`.
+        :param current_block: int - The current block number.
+        :param collateral_change: Decimal - The change in collateral(supply) interest rate.
+        :param debt_change: Decimal - The change in debt(borrow) interest rate.
+        """
+        self.cumulative_collateral[token_name] += collateral_change
+        self.cumulative_debt[token_name] += debt_change
+        self.token_timestamps[token_name] = self.current_timestamp
+        self.current_block = current_block
+
+    def _fill_state_data(self) -> None:
+        """General function for filling the state data."""
+        self._fill_cumulative_data()
+        self._fill_timestamps()
+
+    def _fill_cumulative_data(self) -> None:
+        """Fill the cumulative collateral and debt data with latest block data or default values. Default value is 1."""
+        if self.last_block_data:
+            self.cumulative_collateral = {
+                token_name: Decimal(value) for token_name, value in self.last_block_data.collateral.items()
+            }
+            self.cumulative_debt = {
+                token_name: Decimal(value) for token_name, value in self.last_block_data.debt.items()
+            }
+        else:
+            self.cumulative_collateral = {
+                token_name: Decimal("1") for token_name in TOKEN_MAPPING.values()
+            }
+            self.cumulative_debt = self.cumulative_collateral.copy()
+
+    def _fill_timestamps(self) -> None:
+        """Fill the token timestamps with latest block timestamp or default value. Default value is 0"""
+        if self.last_block_data:
+            self.token_timestamps = {
+                token_name: self.last_block_data.timestamp for token_name in TOKEN_MAPPING.values()
+            }
+        else:
+            self.token_timestamps = {
+                token_name: 0 for token_name in TOKEN_MAPPING.values()
+            }
+
+    def build_interest_rate_model(self, protocol_id: ProtocolIDs) -> InterestRate:
+        """
+        Build the InterestRate model object from the current state data.
+        :param protocol_id: ProtocolIDs - The ID of protocol.
+        :return: InterestRate - The InterestRate model object.
+        """
+        return InterestRate(
+            block=self.current_block,
+            timestamp=self.current_timestamp,
+            protocol_id=protocol_id,
+            **self._serialize_cumulative_data()
+        )
+
+    def _serialize_cumulative_data(self) -> dict[str, dict[str, float]]:
+        """Serialize the cumulative collateral and debt data to write to the database."""
+        collateral = {
+            token_name: float(value) for token_name, value in self.cumulative_collateral.items()
```

Comment:
Is it possible to write decimals instead of floats to the db? I'd like to keep the highest precision possible for the cumulative interest rates.